packages/trader/src/Stores/Modules/Trading/trade-store.ts

```diff
@@ -1221,6 +1317,26 @@ export default class TradeStore extends BaseStore {
             }
             if (this.is_accumulator) this.resetAccumulatorData();
 
+            // Sometimes the initial barrier doesn't match with current barrier choices received from API.
+            // When this happens we want to populate the list of barrier choices to choose from since the value cannot be specified manually
+            if (this.is_vanilla) {
+                const { barrier_choices, max_stake, min_stake } = response.error.details ?? {};
+                this.setStakeBoundary(contract_type, min_stake, max_stake);
+                this.setStrikeChoices(barrier_choices as string[]);
+                if (!this.strike_price_choices.includes(this.barrier_1)) {
+                    // Since on change of duration `proposal` API call is made which returns a new set of barrier values.
+                    // The new list is set and the mid value is assigned
+                    const index = Math.floor(this.strike_price_choices.length / 2);
+                    this.barrier_1 = this.strike_price_choices[index];
+                    this.onChange({
+                        target: {
+                            name: 'barrier_1',
+                            value: this.barrier_1,
+                        },
+                    });
+                }
+            }
+
             // Sometimes when we navigate fast, `forget_all` proposal is called immediately after proposal subscription calls.
```

Comment:
i noticed you removed this code block in turbos pr. 

should i remove this too?